LFRIC/docker/README.md

```diff
@@ -22,34 +27,164 @@ ENV FFLAGS "-I$BUILD_DIR/XIOS/inc -I$INSTALL_DIR/include -I$INSTALL_DIR/mod"
+ && ./install_lfric_env.sh \
+ && ./install_xios_env.sh
+```
+The scripts that build the libraries are provided in the repository: 
+- [install_lfric_env.sh](https://github.com/eth-cscs/ContainerHackathon/blob/master/LFRIC/docker/install_lfric_env.sh) sets up the environment and builds the dependencies of `LFRic` without the `XIOS`;
+- [install_xios_env.sh](https://github.com/eth-cscs/ContainerHackathon/blob/master/LFRIC/docker/install_xios_env.sh) creates architecture files needed to build `XIOS` and builds `XIOS`.
+
+## Library versions and settings
+
+* All libraries were built dynamically (the `install_lfric_env.sh` script contains commented out instructions for a static build if required).
+
+* The `MPICH` version used in the current Met Office (MO) `LFRic` build system is 3.3. Here we used 3.1.4 for compatibility with Piz Daint libraries.
+
+* `HDF5`, `NetCDF`, `NetCDF-Fortran`, `NetCDF-C++` and `pFUnit` are also slighlty older than in the current MO `LFRic` build system, whereas `YAXT` is the same version.
+
+* [`XIOS`](https://forge.ipsl.jussieu.fr/ioserver) is a tricky beast to build as not every revision/release will work with every compiler and/or every compiler release.
+  For instance, the MO GCC 6.1.0 environment uses revision 1537 which is too old for GCC 7.4.0 used here. Unfortunately, the current `XIOS` trunk produces a build which
+  segfaults at runtime for the GCC release and other libraries used in this container. There is, however, a relatively recent revision before the segfault bug that
+  was appropriate for this container.
+
+* Here we used the same `PSyclone` release (1.7.0) that is used by the current `LFRic` trunk (as of 4 December 2019), built with Python 2 environment (the move to
+  Python 3 and newest `PSyclone` is in progress). Not every `PSyclone` release will work with every `LFRic` trunk revision. The `LFRic` - `PSyclone` compatibility
+  table is give in this [`LFRic` wiki (requires login)](https://code.metoffice.gov.uk/trac/lfric/wiki/LFRicTechnical/VersionsCompatibility).
+
+## Tips & tricks
+
+* [Sarus](https://user.cscs.ch/tools/containers/sarus) on Piz Daint had issues with locating `MPICH`. Hence, the lines in the first
+  Dockerfile above
+  ```
+  # Adds config file for MPICH for Sarus on Piz Daint
+  RUN echo "/usr/local/src/gnu_env/usr/lib" > /etc/ld.so.conf.d/mpich.conf \
+   && ldconfig
+  ```
+  were aded to help locate MPICH libraries (see e.g. [this link])(https://unix.stackexchange.com/questions/425251/using-ldconfig-and-ld-so-conf-versus-ld-library-path). 
+  Note, this would normally be achieved by building Docker container with the option
+  `--add-hostname $HOSTNAME:127.0.0.1` to `docker build` command. Alternatively it could be done by modifying the `/etc/hosts` file
+  in the Docker container as described [here](https://stackoverflow.com/questions/23112515/mpich2-gethostbyname-failed/23118973)
+  and the committing the change. However, neither of these solutions worked here.
```

Comment:
I would rephrase the paragraph as follows: 
* The container tool [Sarus](https://user.cscs.ch/tools/containers/sarus) supported on Piz Daint can add the proper hook to the host MPI library if the command `ldconfig` has been run to configure dynamic linker run-time bindings . Since we have installed the `MPICH` library in a non-default location, the command `ldconfig` would not be able to find the library in the custom path within the container. Therefore,  before running `ldconfig` we added the path of MPICH  to `/etc/ld.so.conf.d/mpich.conf` as in the example below:
 ```
  # Adds config file for MPICH for Sarus on Piz Daint
  RUN echo "/usr/local/src/gnu_env/usr/lib" > /etc/ld.so.conf.d/mpich.conf \
   && ldconfig
  ```
Please have a look [here](https://unix.stackexchange.com/questions/425251/using-ldconfig-and-ld-so-conf-versus-ld-library-path) for more information. 
 
*  Some libraries (e.g. `YAXT`) perform a test run of a minimal MPI code during the configure step of the installation procedure: the test might fail within the Docker container if the local hostname cannot be resolved correctly. In order to do that, the local hostname should be available in the file `/etc/hosts`, which can be achieved adding the option `--add-hostname $HOSTNAME:127.0.0.1`  to `docker build` command. 
If this solution does not work, one needs to edit the `/etc/hosts` file of the Docker container directly using `docker run` and commit the change with `docker commit`, since editing the `/etc/hosts' file is not possible within the Dockerfile.
For more details, please check [this link](https://stackoverflow.com/questions/23112515/mpich2-gethostbyname-failed/23118973).